src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.number
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(defn real?
+  [x]
+  (and (number? x)
+       (case x
+         Double/POSITIVE_INFINITY false
+         Double/NEGATIVE_INFINITY false
+         Double/NaN false
+         true)))
+
+(s/def ::real
+  (s/spec real?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
I'll mull over `finite?`. Note of course there are plenty of real, irrational numbers that do pass this predicate: `(/ 1 3)` and friends. It's not pi's fault that clojure doesn't have an e.g. Taylor series expansion type to compute it to arbitrary precisions :)

If the predicate failed to cover such a case, I'd buy the argument that it's not named correctly. But AFAICT, it covers all real numbers expressible in clojure's arithmetic.